include/villas/fpga/utils.hpp

```diff
@@ -17,11 +15,120 @@ namespace fpga {
 std::shared_ptr<fpga::PCIeCard>
 setupFpgaCard(const std::string &configFile, const std::string &fpgaName);
 
-void configCrossBarUsingConnectString(std::string connectString,
-	std::shared_ptr<villas::fpga::ip::Dma> dma,
-	std::vector<std::shared_ptr<fpga::ip::AuroraXilinx>>& aurora_channels);
-
 void setupColorHandling();
 
+class ConnectString {
+public:
+	ConnectString(std::string& connectString, int maxPortNum = 7);
+
+	void parseString(std::string& connectString);
+	int portStringToInt(std::string &str) const;
+
+	void configCrossBar(std::shared_ptr<villas::fpga::ip::Dma> dma,
```

Comment:
`dma->connectLoopback()`
is a method of Dma so we need to pass Dma. The ConnectString is supposed to connect Aurora Channels. I want to keep it simple for the user and not have them specify unique IP names. This motivates passing AuroraXilinx instead of Node.
A more generic ConnectString Parser could make sense but was not what I needed here.